napalm/ios/ios.py

```diff
@@ -901,6 +903,16 @@ def get_lldp_neighbors_detail(self, interface=""):
         if len(lldp_entries) == 0:
             return {}
 
+        # format chassis_id for consistency, if it is a mac address
+        for entry in lldp_entries:
+            if re.search(
+                r"^(\d|\w){4}.(\d|\w){4}.(\d|\w){4}$", entry["remote_chassis_id"]
```

Comment:
Instead of doing this regex check, I'd just apply the `convert` function instead, e.g.,

```suggestion
entry["remote_chassis_id"] = napalm.base.helpers.conver(napalm.base.helpers.mac, entry["remote_chassis_id"], entry["remote_chassis_id"])
```

Reply:
Thanks for the quick review and all the suggestions. Just checking: I agree with your suggestion, but if the chassis id cannot be converted, an empty chassis id will be returned. This is an example value from a chassis id from one of the tests:
```
In [75]: id = "hmbbkku-la"

In [76]: id = napalm.base.helpers.convert(napalm.base.helpers.mac, id)

In [77]: id
Out[77]: ''
```

Is that OK?

Reply:
I have addressed the comment with the latest commit. Let me know what you think @mirceaulinic

Reply:
Yes, there should be a third argument, with the default, notice in the suggestion above:

```python
entry["remote_chassis_id"] = napalm.base.helpers.conver(napalm.base.helpers.mac, entry["remote_chassis_id"], entry["remote_chassis_id"])
```

When unable to convert to the normalised MAC format, it'd default to the extracted value.

Reply:
Thank you for the help! Added.